### PR TITLE
Replace forbidden characters in the incoming filename on Windows...

### PR DESCRIPTION
**Pull request by redmanmale**
with underscores.

Both for user-accepted and auto-accepted files (#611).

---
This change is [Reviewable](https://reviewable.io/reviews/utox/utox/919)

### CODE REVIEW

File: src/windows/main.7.c [hunk 1]
```diff
@@ -55,13 +55,32 @@ void native_export_chatlog_init(uint32_t friend_number) {
     }
 }
 
+void sanitize_filename(uint8_t * filename) {
```
Comment (robinlinden) [inline src/windows/main.7.c]: `*filename`

**Comment by redmanmale** [comment]
> Move it to a file you can include from both main.xp.c and main.7.c.

How about src/windows/utf8.c?

> Are there any utf8 characters we need to check for?

At msdn [article](https://msdn.microsoft.com/en-us/library/windows/desktop/aa365247(v=vs.85).aspx) about naming conventions there is number of forbidden filenames (COM1, COM2, etc), but I don't want to check for all of them.

> Can we get utf8 filename here?

What do you mean? Isn't `file->name` (char*) already utf8?

**Comment by GrayHatter** [comment]
Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion.

---

*[src/windows/main.7.c, line 58 at r1](https://reviewable.io:443/reviews/utox/utox/919#-KkXcBMCPw267BjQiRLV-r1-58:-KkckUzqwDHIcalvRa5c:bha599a) ([raw file](https://github.com/utox/utox/blob/6ee053ff567d14d694cb4942017c5b9b94a08399/src/windows/main.7.c#L58)):*
<details><summary><i>Previously, robinlinden (Robin Lindén) wrote…</i></summary><blockquote>

`*filename`
</blockquote></details>

this should be

```
// TODO, add utf8 support
if ( strchr("<>:/|?*\\\"', filename[i]) {
    ...
```

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/utox/utox/919)*

**Comment by redmanmale** [comment]
Review status: 0 of 4 files reviewed at latest revision, 1 unresolved discussion.

---

*[src/windows/main.7.c, line 58 at r1](https://reviewable.io:443/reviews/utox/utox/919#-KkXcBMCPw267BjQiRLV-r1-58:-Kkgky2Ndfq_lSc4tMi7:b-896fix) ([raw file](https://github.com/utox/utox/blob/6ee053ff567d14d694cb4942017c5b9b94a08399/src/windows/main.7.c#L58)):*
<details><summary><i>Previously, GrayHatter (Gregory Mullen) wrote…</i></summary><blockquote>

this should be

```
// TODO, add utf8 support
if ( strchr("<>:/|?*\\\"', filename[i]) {
    ...
```
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/utox/utox/919)*

File: src/windows/main.7.c (hunk 1 above)
Comment (redmanmale) [inline src/windows/main.7.c]: Done.

**Comment by GrayHatter** [comment]
Reviewed 4 of 4 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[src/windows/utf8.c, line 16 at r2](https://reviewable.io:443/reviews/utox/utox/919#-KkmNO2XrA3o6XbNPq8z:-KkmNO2XrA3o6XbNPq9-:b-ha9zmy) ([raw file](https://github.com/utox/utox/blob/24af07e4f9d5fb0cf1620935a2e3b8181acde668/src/windows/utf8.c#L16)):*
> ```C
> void sanitize_filename(uint8_t *filename) {
>     for (size_t i = 0; filename[i] != '\0'; ++i) {
>         if ((filename[i] >= 1 && filename[i] <= 31) || strchr("<>:\"/\\|?*", filename[i])) {
> ```

This should return false, and the caller should abort. If `filename[i]` is ever less than 32, that's a malicious filename, and it's safe to exit early.

If you want to mark that as OOS open an issue for it, and I'll approve this pull

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/utox/utox/919)*

**Comment by endoffile78** [comment]
Reviewed 4 of 4 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/utox/utox/919)*

**Comment by GrayHatter** [comment]
:lgtm:

**Comment by endoffile78** [comment]
:lgtm:

---

Reviewed 4 of 4 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/utox/utox/919#-:-Kkpjfy-BJ7ruYWXd3Wt:bnfp4nl)*

**Comment by redmanmale** [comment]
@robinlinden 
Could you approve this PR via Reviewable too?

**Comment by robinlinden** [comment]
Of course. Sorry, reviewed on my phone and forgot to do it on here too. :lgtm:

---

Reviewed 4 of 4 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/utox/utox/919#-:-KkqjMBiSjURsjwiqbqJ:b-y3nra4)*

**Comment by GrayHatter** [comment]
cleared my comment, :lgtm:

**Pull request merged**